Source/WebCore/page/Page.cpp

```diff
@@ -3276,6 +3276,15 @@ void Page::setVisitedLinkStore(Ref<VisitedLinkStore>&& visitedLinkStore)
     invalidateStylesForAllLinks();
 }
 
+uint64_t Page::noiseInjectionHashSaltForDomain(const RegistrableDomain& domain)
+{
+    if (auto salt = m_noiseInjectionHashSalts.get(domain))
+        return salt;
+    uint64_t salt = cryptographicallyRandomNumber<uint64_t>();
+    m_noiseInjectionHashSalts.set(domain, salt);
+    return salt;
```

Comment:
Nit - it would be cleaner (and more efficient) to use `m_noiseInjectionHashSalts .ensure()` here (and return the iterator value).